server/webhook.go

```diff
@@ -19,6 +21,12 @@ const (
 )
 
 func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+
+	if subtle.ConstantTimeCompare([]byte(p.configuration.WebhookSecret), []byte(params["webhooksecret"])) == 0 {
```

Comment:
1. Can we use `p.getConfiguration()` here instead of `p.configuration`?

1. We should also make sure `p.configuration.WebhookSecret` is not an empty string here.

1. We should put a check in `config.IsValid` to ensure it's set there as well.
https://github.com/mattermost/mattermost-plugin-bitbucket/blob/4869666f42892c41eb21d98747fa4f9b8552d7ea/server/configuration.go#L37